packages/flutter/lib/src/widgets/reorderable_list.dart

```diff
@@ -802,40 +802,29 @@ class SliverReorderableListState extends State<SliverReorderableList> with Ticke
   }
 
   void _dragEnd(_DragInfo item) {
-    // No changes required if last child is being inserted into the last position.
-    if ((_insertIndex! + 1 == _items.length) && _reverse) {
-      final RenderBox lastItemRenderBox =  _items[_items.length - 1]!.context.findRenderObject()! as RenderBox;
-      final Offset lastItemOffset =  lastItemRenderBox.localToGlobal(Offset.zero);
-
-      // When drag starts, the corresponding element is removed from
-      // the list, and moves inside of [ReorderableListState.CustomScrollView],
-      // which gives [CustomScrollView] a variable height.
-      //
-      // So when the element is moved, delta would change accordingly,
-      // and since it's the last element,
-      // we animate it back to it's position and add it back to the list.
-      final double delta = item.itemSize.height;
-
-      setState(() {
-        _finalDropPosition = Offset(lastItemOffset.dx, lastItemOffset.dy  - delta);
-      });
-      return;
-    }
     setState(() {
-      if (_insertIndex == item.index) {
-        _finalDropPosition = _itemOffsetAt(_insertIndex! + (_reverse ? 1 : 0));
-      } else if (_insertIndex! < widget.itemCount - 1) {
-        // Find the location of the item we want to insert before
-        _finalDropPosition = _itemOffsetAt(_insertIndex!);
+      if (_reverse) {
+        if (_insertIndex! == item.index) {
```

Comment:
Since this same case is repeated below on 819 in the following else, can this instead be reworked to contain the `_insertIndex! == item.index` case outside of whether or not it is reversed? Might be easier to follow then

Reply:
The handling of `_insertIndex! == item.index` is different depending on the value of `_reverse`. In the case where `_reverse` is true, we offset the item's own extent, but in the case where `_reverse` is false, we don't need to do this. Therefore, it's not possible to take this case out of the _reverse condition. I believe the current implementation is the most straightforward way to handle these different scenarios.

Reply:
I still find the code very difficult to follow. The prior implementation incorporated reverse in the if clauses or when making the function calls. It is hard to tell what has actually changed here.

Reply:
Got it, I fixed some code, please continue the review